### PR TITLE
Fix Bug 1153588, support SeaMonkey on plugin check

### PR DESCRIPTION
**Pull request by schalkneethling**
@alexgibson r?

### CODE REVIEW

File: media/js/test/spec/global.js [hunk 1]
```diff
@@ -320,6 +320,51 @@ describe('global.js', function() {
...
+
+        it('should consider SeaMoneky as SeaMonkey', function() {
+            var ua = 'Mozilla/5.0 (Windows NT 6.1; WOW64; rv:40.0) Gecko/20100101 Firefox/40.0 SeaMonkey/2.37a1';
+            var result = isSeaMonkey(ua);
+            expect(result).toBeTruthy();
+        });
+
+        it('should not consider Firefox to be SeaMonkey', function() {
+            var ua = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.8; rv:23.0) Gecko/20100101 Firefox/23.0';
+            var result = isFirefox(ua);
```
Comment (alexgibson) [inline media/js/test/spec/global.js]: This should be testing `isSeaMonkey()`, right? Same for occurrences below.

Comment (schalkneethling) [inline media/js/test/spec/global.js]: Oooooh ouch, copy/paste #fail

File: media/js/base/global.js [hunk 2]
```diff
@@ -98,6 +98,11 @@ function isFirefox(userAgent) {
     );
 }
 
+function isSeaMonkey(userAgent) {
+    var ua = userAgent || navigator.userAgent;
+    return ((/SeaMonkey/i).test(ua));
```
Comment (alexgibson) [inline media/js/base/global.js]: Nit: all the brackets might not be required here: could this just be `return /SeaMonkey/i.test(ua);`?

**Comment by alexgibson** [comment]
This looks good to me, thanks @schalkneethling. I think the function `isSeaMonkey` is probably ok to leave in `global.js` for now. It is likely pretty uncommon.

In the near future I would like to factor out all these UA detection methods into their own lib for consumption in bedrock, instead of hanging them off global scope everywhere. But this is a larger change to the code base and probably outside the scope of this bug. 

I think this is fine for the time being. Thanks and r+wc

File: media/js/test/spec/global.js [hunk 3]
```diff
@@ -320,6 +320,51 @@ describe('global.js', function() {
         });
     });
 
+    describe('isSeaMonkey', function() {
+
+        it('should consider SeaMoneky as SeaMonkey', function() {
```
Comment (stephendonner) [inline media/js/test/spec/global.js]: Uber (unsolicited) nit: s/SeaMoneky/SeaMonkey

Comment (schalkneethling) [inline media/js/test/spec/global.js]: bugger, nice catch @stephendonner ;)

**Pull request merged**